src/wallet/coinselection.cpp

```diff
@@ -87,13 +87,18 @@ std::optional<SelectionResult> SelectCoinsBnB(std::vector<OutputGroup>& utxo_poo
     std::vector<size_t> best_selection;
     CAmount best_waste = MAX_MONEY;
 
+    // Check if the current feerate is expensive.  If it is expensive, we minimize the size of
+    // the result set and thus minimizing the transaction fee.  If the fees are not expensive,
+    // we maximize the result set to reduce the amount of UTXOs (fragmentation).
```

Comment:
My comment was misleading it looks like.  It's true that a true maximum or true min may not be found before the iteration limit.  Also, I appreciate you pointing out that how many or how few UTXO's are targeted can depend on how much the longterm and short term fees differ.  Perhaps we can just do without a comment here since the statement is self explanatory.  I agree that further comment on how the selection process works is better to place on the function.